ui/imports/shared/popups/CommunityMembershipSetupDialog.qml

```diff
@@ -159,18 +168,25 @@ StatusStackModal {
 
         readonly property int selectedSharedAddressesCount: d.selectedSharedAddressesMap.size
 
+        readonly property int eligibleToJoinAs: {
+            d.selectedSharedAddressesMap // needed for binding
+            return PermissionsHelpers.isEligibleToJoinAs(root.permissionsModel)
+        }
+
         property var initialAddressesModel: SortFilterProxyModel {
             sourceModel: root.walletAccountsModel
             sorters: [
-                ExpressionSorter {
-                    function isGenerated(modelData) {
-                        return modelData.walletType === Constants.generatedWalletType
+                FastExpressionSorter {
+                    function isGenerated(walletType) {
+                        return walletType === Constants.generatedWalletType
                     }
 
                     expression: {
-                        return isGenerated(modelLeft)
+                        return isGenerated(modelLeft.walletType)
                     }
+                    expectedRoles: ["walletType"]
                 },
+                // FIXME sort by balances
```

Comment:
Still needed?

Reply:
Yeah, the sorting will be different here (followup PRs)